servers/rendering/shader_compiler.cpp

```diff
@@ -1320,12 +1321,39 @@ Error ShaderCompiler::compile(RS::ShaderMode p_mode, const String &p_code, Ident
 	info.shader_types = ShaderTypes::get_singleton()->get_types();
 	info.global_variable_type_func = _get_variable_type;
 
+	// Cheat the context here for now.
```

Comment:
This code __really__ should not be here, the compiler should give you the right file and line when there is an error, and it should not depend on the preprocessor at this point.